### PR TITLE
DOCS-7907 Add IDE Plugins to Code Analysis Navigation

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do? What is the motivation?

Adds IDE Plugins section to the Code Analysis TOC.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/developers/ide_plugins/visual_studio/_index.md [hunk 1]
```diff
@@ -37,6 +37,16 @@ Analyze and improve the performance of your applications with real-time profilin
 
 {{< img src="/developers/ide_plugins/visual_studio/top-list.png" alt="The Code Insights view">}}
 
+### Static Analysis
+
+The Datadog extension runs [Static Analysis][14] rules on the source files you have open in your Solution. The goal is to detect and fix problems such as maintainability issues, bugs, or security vulnerabilities in your code before you commit your changes.
+
+Static Analysis supports scanning across multiple programming languages. For a complete list, see [Static Analysis Rules][15]. For file types belonging to supported languages, issues are shown in the source code editor using Visual Studio's code inspection system, and suggested fixes can be applied directly:
+
+{{< img src="/developers/ide_plugins/visual_studio/TBD.png" alt="A static analysis rule violation and recommended fix." style="width:100%;" >}}
```
Comment (alai97) [inline content/en/developers/ide_plugins/visual_studio/_index.md:46]: To be replaced: image in dark mode @markazerdd

File: config/_default/menus/main.en.yaml [hunk 2]
```diff
@@ -3742,11 +3742,16 @@ menu:
       parent: code_analysis
       identifier: code_analysis_github_prs
       weight: 4
+    - name: IDE Plugins
+      url: developers/ide_plugins
+      parent: code_analysis
+      identifier: code_analysis_ide_plugins
+      weight: 5
```
Comment (hestonhoffman) [inline config/_default/menus/main.en.yaml:3749]: Have you tried a [full-page reuse](https://datadoghq.atlassian.net/wiki/spaces/docs4docs/pages/1182958054/Markdown#Reusing-a-full-page) here? Might be a good use case for it.

**Review by hestonhoffman — APPROVED** [review]
LGTM. Left a suggestion for content reuse

File: content/en/developers/ide_plugins/visual_studio/_index.md (hunk 1 above)
Comment (alai97) [reply content/en/developers/ide_plugins/visual_studio/_index.md:46]: @AlonAm to add the screenshot in here

Comment (AlonAm) [reply content/en/developers/ide_plugins/visual_studio/_index.md]: Added a new screenshot, let me know what you think, thanks

**Comment by jfree** [comment]
Hi all, Alon brought this PR to my attention.  I had another (related) PR https://github.com/DataDog/documentation/pull/23376 approved and merged yesterday and it will cause a conflict with this one so you'll want to update to `master` and check.

**Review by markazerdd — APPROVED** [review]

File: config/_default/menus/main.en.yaml (hunk 2 above)
Comment (alai97) [reply config/_default/menus/main.en.yaml:3869]: Thanks for the suggestions! I think I'll create some dedicated pages in the Code Analysis doc set that are tailored to Code Analysis-specific interactions with the IDE plugins.

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**